python/ray/rllib/env/external_multi_agent_env.py

```diff
@@ -0,0 +1,262 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from six.moves import queue
+import threading
+import uuid
+
+from ray.rllib.utils.annotations import PublicAPI
+
+
+@PublicAPI
+class ExternalMultiAgentEnv(threading.Thread):
```

Comment:
Should this extend ExternalEnv?

Reply:
The only differences I see between ExternalEnv and ExternalMultiAgentEnv is parameter types in the method signatures (e.g. action vs. action_dict), so this should be ok?